File: src/main/java/zoo/api/service/animal/AnimalService.java
package zoo.api.service.animal;

import zoo.api.bo.animal.Animal;

import java.util.List;

public interface AnimalService {
    List<Animal> findAllAnimals();

    Animal findAnimalById(int id);

    Animal addAnimal(Animal animal);

    Animal updateAnimal(Animal animal);

    void deleteAnimal(Animal animal);
}

File: zoo-ui/src/services/AnimalService.js
import axios from 'axios';

const ANIMAL_API_BASE_URL = "http://localhost:8080/zoo-api/animals/";

class AnimalService {

    getAnimals(){
        return axios.get(ANIMAL_API_BASE_URL);
    }

    createAnimal(animal){
        return axios.post(ANIMAL_API_BASE_URL, animal);
    }

    getAnimalById(animalId){
        return axios.get(ANIMAL_API_BASE_URL + animalId);
    }

    updateAnimal(animal, animalId){
        return axios.put(ANIMAL_API_BASE_URL + animalId, animal);
    }

    deleteAnimal(animalId){
        return axios.delete(ANIMAL_API_BASE_URL + animalId);
    }
}

export default new AnimalService()
